ako-gateway-api/k8s/validator.go

```diff
@@ -166,8 +157,8 @@ func isValidListener(key string, gateway *gatewayv1.Gateway, gatewayStatus *gate
 	for _, gatewayInNamespace := range gatewayNsList {
 		if gateway.Name != gatewayInNamespace.Name {
 			for _, gwListener := range gatewayInNamespace.Spec.Listeners {
-				if *listener.Hostname == *gwListener.Hostname || isRegexMatch(string(*listener.Hostname), string(*gwListener.Hostname), key) || isRegexMatch(string(*gwListener.Hostname), string(*listener.Hostname), key) {
-					utils.AviLog.Errorf("key: %s, msg: Hostname overlaps or is same as an existing gateway %s hostname %s", key, gatewayInNamespace.Name, *gwListener.Hostname)
+				if *listener.Hostname == *gwListener.Hostname {
```

Comment:
We should remove this condition alltogether

Reply:
I didn't get intention of removing it. If other gateway as the same hostname, we shouldn't process that gateway as same fqdn can't be associated with different vips.

Reply:
Please check yaml for this conformance test https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/tests/gateway-secret-reference-grant-all-in-namespace.go. This wont pass unless this check is removed

Reply:
Also, if one of the listener is invalid, as per today's discussion we should partially accept the gateway and set that listener status to be unaccepted

Reply:
For this change, we need to merge the gateways. This will major change as it will involve
1. Uniquely identifying all valid listeners, clubbing those together. 
2. Associating those gateway with single Parent VS. So it will introduce naming convention changes.
3. Associating routes with common parent VS.

We can take this up as part of another PR. Please raise Jira for that.

Reply:
https://avinetworks.atlassian.net/browse/AV-212611